pkg/openstack/nova.go

```diff
@@ -48,6 +49,111 @@ func ReconcileNova(ctx context.Context, instance *corev1beta1.OpenStackControlPl
 		return ctrl.Result{}, nil
 	}
 
+	// Create service overrides to pass into the service CR
+	// and expose the public endpoint using a route per default.
+	// Any trailing path will be added on the service-operator level.
+	apiServiceOverrides := map[string]service.OverrideSpec{}
+	serviceDetails := []ServiceDetails{}
+
+	for _, endpointType := range []service.Endpoint{service.EndpointPublic, service.EndpointInternal} {
+		sd := ServiceDetails{
+			ServiceName:         novav1.GetAPIServiceName(),
+			Namespace:           instance.Namespace,
+			Endpoint:            endpointType,
+			ServiceOverrideSpec: instance.Spec.Nova.Template.APIServiceTemplate.Override.Service,
+			RouteOverrideSpec:   instance.Spec.Nova.APIOverride.Route,
+		}
+
+		svcOverride, ctrlResult, err := sd.CreateRouteAndServiceOverride(ctx, instance, helper)
+		if err != nil {
+			return ctrlResult, err
+		} else if (ctrlResult != ctrl.Result{}) {
+			return ctrlResult, nil
+		}
+
+		serviceDetails = append(
+			serviceDetails,
+			sd,
+		)
+
+		apiServiceOverrides[string(endpointType)] = *svcOverride
+	}
+
```

Comment:
I think you missed the case when NovaMetadata is deployed not per cell but with one single service on the top level. While NovaMetadata does not need to be route to the k8s public network, I think the service overrides still need to be passed through. (If I'm mistaken then I don't get while you have explicit NovaMetadat handling below for the case where NovaMetadata deployed per cell)